packages/ingest/FtpProviderClient.js

```diff
@@ -74,38 +93,44 @@ class FtpProviderClient {
     const client = await this.buildFtpClient();
 
     return new Promise((resolve, reject) => {
-      client.on('error', reject);
+      client.on('error', this.errorHandler.bind(this, reject));
       client.get(remotePath, localPath, (err) => {
-        client.destroy();
-
-        if (err) reject(err);
-        else {
-          log.info(`Finishing downloading ${remoteUrl}`);
-          resolve(localPath);
+        if (err) {
+          return this.errorHandler(reject, err);
         }
+        log.info(`Finishing downloading ${remoteUrl}`);
+        client.destroy();
+        return resolve(localPath);
       });
     });
   }
 
+  /**
+   * List all files from a given endpoint
+   * @param {string} path - path to list
+   * @return {Promise}
+   * @private
+   */
   async _list(path, _counter = 0) {
     let counter = _counter;
     const client = await this.buildFtpClient();
     return new Promise((resolve, reject) => {
-      client.on('error', reject);
+      client.on('error', this.errorHandler.bind(this, reject));
       client.ls(path, (err, data) => {
-        client.destroy();
         if (err) {
-          if (err.message.includes('Timed out') && counter < 3) {
+          const message = err.message || err.text;
+          if (message && message.includes('Timed out') && counter < 3) {
             log.error(`Connection timed out while listing ${path}. Retrying...`);
             counter += 1;
             return this._list(path, counter).then((r) => {
-              log.info(`${counter} retry suceeded`);
+              log.info(`${counter} retry succeeded`);
               return resolve(r);
-            }).catch((e) => reject(e));
+            }).catch(this.errorHandler.bind(this, reject));
```

Comment:
So the net improvement of moving to the new error handling is improved error messages and ensuring that the client gets destroyed before rejecting?

Reply:
Yeah, it's DRYing up a bunch of additional `client.destroy` calls, and ensuring we throw a standardized error out of the FtpProviderClient.